src/main/java/edu/harvard/iq/dataverse/EditDatafilesPage.java

```diff
@@ -1655,10 +1679,11 @@ public void uploadFinished() {
         // Add the file(s) added during this last upload event, single or multiple, 
         // to the full list of new files, and the list of filemetadatas 
         // used to render the page:
-                
-        if (mode == FileEditMode.CREATE) {
-            ingestService.addFilesToDataset(workingVersion, uploadedFiles);
-        }
+             
+        // Uh, maybe we shouldn't be doing this??
```

Comment:
The commented-out method was adding new files to the dataset, by linking the files to the dataset, and the filemetadatas to the datasetversion. 
The other method, addFiles() does that as well; (in addition to permanently saving the physical files on disk/s3/etc.)
The other method was needed because the CreateDatasetCommand and UpdateDatasetCommand were handling new uploaded files differently.  So because of that, in the former case we needed to attach the files to the dataset before they were saved; and in the latter we were doing it inside the addFiles() method... This different behavior of the 2 commands was ugly and a source of problems (things working when uploading files into an existing dataset, but not working on create; because of the 2 different pathways it was easy to overlook something). 
Plus, for the purposes of this issue specifically, I didn't want to add files to the dataset until I am sure they have been successfully saved. So I moved the saving step out of the Create command. 

I left the commented out method in place mostly for clarity, for when we'll be merging this with the work Michael has done that also involved the CreateDataset workflow. 

But I will remove the "maybe we shouldn't be doing this" - that was for me still trying to figure out what was going on; I meant to get rid of it before making the PR.

Reply:
Thanks Leonid, the info is helpful for my understanding!